sematic/db/models/mixins/has_user_mixin.py

```diff
@@ -2,10 +2,13 @@
 from typing import Optional
 
 # Third-party
-from sqlalchemy import Column, types
+from sqlalchemy import Column, ForeignKey, types
+from sqlalchemy.ext.declarative import declared_attr
 from sqlalchemy.orm import declarative_mixin
 
 
 @declarative_mixin
 class HasUserMixin:
-    user_id: Optional[str] = Column(types.String(), nullable=True)
+    @declared_attr
+    def user_id(cls):
+        return Column(types.String(), ForeignKey("users.id"), nullable=True)
```

Comment:
sqlalchemy needs this to be a `declared_attr` in order to assign a foreign key to a table via a mixin.

Reply:
I wonder if this breaks anything in `JSONEncodableMixin`, since this is our first usage of `declared_attr`s. It'll probably come up in tests if it is a problem, but probably a good idea to confirm that we have tests that validate serializing/deserializing the `user_id` field in one of the object types that uses this mixin.

Reply:
Confirmed the User is correctly sourced and ser/des when accessing a Run through the API.